utreexo-validation-bip.md

```diff
@@ -0,0 +1,328 @@
+as all proofs are with respect to the Merkle roots of the entire set.
+
+For this reason, we define which UTXOs are not inserted to the accumulator.  Any
+variations here will result in Utreexo nodes with incompatible proofs.
+
+##### Provably unspendable transaction outputs
+
+There are outputs in the Bitcoin network that we can guarantee that they cannot
+be spent without a hard-fork of the network. The following output types are not
+added to the accumulator:
+- Outputs that start with an OP_RETURN (0x6a)
+- Outputs with a scriptPubkey larger than 10,000 bytes
+
+##### Same block spends
+
+Often, UTXOs are created and spent in the same block. This is allowed by Bitcoin
+consensus rules as long as the output being spent is created by a transaction earlier
+in the block than the spending transaction.
+In Utreexo, nodes inspect blocks and identify which outputs are being created
+and destroyed in the same block, and exclude them from the accumulator and proofs.
+
+There's no need to provide proofs for outputs which have been created in the same
+block. Adding and then immediately removing the output from the accumulator would be
+possible but doesn't serve any purpose - once outputs are spent their past existence
+cannot be proven with the Utreexo accumulator (and SPV proofs already provide that).
+
+For these reasons, outputs which are spent in the same block where they are created
+are omitted from the accumulator, and those inputs are omitted from block proofs.
+
+#### Order of operations
+
+The Utreexo accumulator lacks associative properties during addition and the
+ordering of which UTXO hash gets added first is consensus critical. For
+the modification of the accumulator the steps are as follows:
+
+1. Batch remove the UTXOs that were spent in the block based on the algorithm
+   defined in BIP-????. Deletions itself are order-independent.
+2. Batch add all non-excluded outputs in the order they're included in the
+   Bitcoin block. Additions are order-dependent.
+
+The removal and the addition of the hashes follow the algorithms defined in
+BIP-????.
+
+#### Format of the UTXO proof
+
+The UTXO proof has 2 elements: the accumulator proof and the leaf data. The
+leaf data provides the necessary UTXO data for block validation that would be
+stored locally for non-Utreexo nodes. The accumulator proof proves that the
+given UTXO hash preimages are committed in the accumulator.
+
+Accumulator proof is defined in BIP-????, and contains two elements:
+
+1. A vector of positions of the UTXO hashes in the accumulator.
+2. A vector of hashes required to hash up to the roots.
+
+For (1), positions are in the order of the leaves that are being proved in
+the accumulator. These are all the inputs in the natural blockchain order that
+excludes the same block spends.
+
+The UTXO hash preimages follow the same ordering as (1) in the accumulator
+proofs. Each of the positions in (1) refer to the UTXO hash preimage in the same
+index.
```

Comment:
For some reason I had thought that the accumulator proof was a Merkle branch, but now reading this, it makes me think that the proofs are built-up from the leaf preimages. Which of the two is correct, and could you perhaps check whether some more clarification should be added here to make it unambiguous? 
This might also just be me mixing up something as I’m trying to puzzle together everything that is going on.